tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/BootstrapTasks/GetBuildTools.cs

```diff
@@ -210,6 +212,26 @@ public override bool Execute()
             return true;
         }
 
+        /// <summary>
+        /// Copy powershell modules to the user profile powershell dir
+        /// </summary>
+        /// <param name="psModulesDir">Powershell modules source dir</param>
+        /// <param name="filesToCopy">List of files to copy to tools dir</param>
+        /// <returns></returns>
+
+        private void CopyPowershellModules(string psModulesDir, string[] filesToCopy)
+        {
+            string userProfilePSModulesPath = Environment.GetEnvironmentVariable("PSModulePath").Split(';').First(p => p.StartsWith(Environment.GetEnvironmentVariable("USERPROFILE")));
+            IEnumerable<string> psModulesToCopy = filesToCopy.Where(p=>p.StartsWith("psModules"))
+                                                             .Where(p=>p.EndsWith(".psm1")||p.EndsWith(".psd1")||p.EndsWith(".ps1"));
+            foreach (var module in psModulesToCopy)
```

Comment:
@dsgouda we are assuming we are going to copy this every single time.
It will be worth to have a verification if the files were copied successfully, if not we should fail fast.

Reply:
@shahabhijeet thanks for pointing this out. Made a couple of useful observations here.
- Since we keep track of files we are unable to copy and throw an exception of any of the file copy fails, we should be safe here
- There are instances where `C:\Users\Username\Documents\WindowsPowershell\Modules` is not a part of the `PSModulePath` in env variables. I added a check to find that and throw an exception
- Lastly, once the psm1 file is copied to any of the `PSModulePath` we don't need to do an explicit `Import-Module` for the path since all of the exported cmdlets actually get imported the moment we copy those files to the proper path, remove the unnecessary `Import-Module` powershell call I was making